src/win/pipe.c

```diff
@@ -880,12 +1090,33 @@ int uv_pipe_connect2(uv_connect_t* req,
                      unsigned int flags,
                      uv_connect_cb cb) {
   uv_loop_t* loop;
-  int err;
+  int err = 0;
   size_t nameSize;
   HANDLE pipeHandle = INVALID_HANDLE_VALUE;
   DWORD duplex_flags;
   char* name_copy;
 
+  int use_uds_pipe;
+
+#if defined(UV__ENABLE_WIN_UDS_PIPE)
+  int uds_file_exists;
+  SOCKET uds_client_fd;
+  struct sockaddr_un uds_addr_bind = {0};
+  struct sockaddr_un uds_addr_real = {0};
+  DWORD uds_dummy_send_cnt = 0;
+
+  /* ConnectEx seems has a bug when using with 'sockaddr_un'.
+   * It seems corrupting stack if no such buffer present on stack.
+   * Looks like needs at least 316 bytes to not overwriting some data onto
+   * valid stack spaces, allocating 512 bytes for 'safety'.
+   *
+   * TODO: It still overflow write to this buffer instead of valid
+   * stack, so it is just a dangerous workaround to write to a controlled
+   * dummy memory instead of causing stack corruption.
+   */
+  char uds_dummy_send_buffer[512] = {0};
```

Comment:
I would say it is unacceptable to merge while this bug is in place

Reply:
Yes, and how do you think about not using `ConnectEx` here, and directly use `connect`.

Reply:
IOCP is supported anywhere else, since UDS is very fast and supported by loval driver, it will not be blocking.

Nevermind, the connect/WSAConnect also corrupts memory I don't know why yet.